lib/dry/logic/result.rb

```diff
@@ -56,6 +56,19 @@ def success?
         end
       end
 
+      class Result::LazyValue < Result
+        def to_ary
+          [:input, [rule.name, input, [rule.to_ary]]]
+        end
+        alias_method :to_a, :to_ary
+
+        def input
+          # I think that it's better to return @input instead of nil.
+          # It will demonstrate which input doesn't have appropriate key or attribute
+          success? ? rule.evaluate_input(@input) : nil
```

Comment:
@solnic What do you think about the comment above? Should we return nil in case of failure?